src/cli/cli.cpp

```diff
@@ -4862,17 +4862,20 @@ void Interpreter::ProcessLine(char *aBuf)
     Arg            args[kMaxArgs];
     uint8_t        argsLength;
     const Command *command;
-    otError        error;
+    otError        error = OT_ERROR_NONE;
+
+    OT_ASSERT(aBuf != nullptr);
 
-    VerifyOrExit(aBuf != nullptr && StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1);
+    VerifyOrExit(StringLength(aBuf, kMaxLineLength) <= kMaxLineLength - 1, error = OT_ERROR_PARSE);
 
-    VerifyOrExit(Utils::CmdLineParser::ParseCmd(aBuf, argsLength, args, kMaxArgs) == OT_ERROR_NONE,
-                 OutputLine("Error: too many args (max %d)", kMaxArgs));
+    SuccessOrExit(error = Utils::CmdLineParser::ParseCmd(aBuf, argsLength, args, kMaxArgs));
     VerifyOrExit(argsLength >= 1);
 
 #if OPENTHREAD_CONFIG_DIAG_ENABLE
-    VerifyOrExit((!otDiagIsEnabled(mInstance) || (args[0] == "diag")),
-                 OutputLine("under diagnostics mode, execute 'diag stop' before running any other commands."));
+    VerifyOrExit((!otDiagIsEnabled(mInstance) || (args[0] == "diag")), {
```

Comment:
How about converting this from `VerifyOrExit()` to an `if` (with `ExitNow()` at end of its `{ }` block)? I feel makes it easier to read (specially since we now need a block `{ }` as a `VerifyOrExit()` arg.

```c++
if (otDiagIsEnabled(mInstance) && (args[0] != "diag"))
{
    OutputLine("under diagnostics mode, execute 'diag stop' before running any other commands.");
    ExitNow(error = OT_ERROR_INVALID_STATE);
}
```